docker.go

```diff
@@ -46,9 +46,17 @@ type Parameters struct {
 
 // Volume is the docker volume details associated to the container
 type Volume struct {
-	ContainerPath string `json:"containerPath,omitempty"`
-	HostPath      string `json:"hostPath,omitempty"`
-	Mode          string `json:"mode,omitempty"`
+	ContainerPath  string          `json:"containerPath,omitempty"`
+	HostPath       string          `json:"hostPath,omitempty"`
+	ExternalVolume *ExternalVolume `json:"external,omitempty"`
+	Mode           string          `json:"mode,omitempty"`
+}
+
+// An external volume definition
+type ExternalVolume struct {
+	Name     string             `json:"name,omitempty"`
+	Provider string             `json:"provider,omitempty"`
```

Comment:
I suppose neither `Name` nor `Provider` may be empty strings from an API point of view?

In case they may, we need to pointerize the strings so that `omitempty` does not drop them.